internal/manifests/collector/collector.go

```diff
@@ -51,9 +51,12 @@ func Build(params manifests.Params) ([]client.Object, error) {
 		manifests.Factory(MonitoringService),
 		manifests.Factory(Ingress),
 	}...)
-	if params.OtelCol.Spec.Observability.Metrics.EnableMetrics && featuregate.PrometheusOperatorIsAvailable.IsEnabled() {
+	if params.OtelCol.Spec.Observability.Metrics.EnableMetrics && featuregate.PrometheusOperatorIsAvailable.IsEnabled() && params.OtelCol.Spec.Mode != v1alpha1.ModeSidecar {
 		manifestFactories = append(manifestFactories, manifests.Factory(ServiceMonitor))
 	}
+	if params.OtelCol.Spec.Observability.Metrics.EnableMetrics && featuregate.PrometheusOperatorIsAvailable.IsEnabled() && params.OtelCol.Spec.Mode == v1alpha1.ModeSidecar {
+		manifestFactories = append(manifestFactories, manifests.Factory(PodMonitor))
+	}
```

Comment:
Can we change this to a nested conditional? Like:

```go
if params.OtelCol.Spec.Observability.Metrics.EnableMetrics && featuregate.PrometheusOperatorIsAvailable.IsEnabled() {
  if params.OtelCol.Spec.Mode == v1alpha1.ModeSidecar {
    manifestFactories = append(manifestFactories, manifests.Factory(PodMonitor))
  } else {
    manifestFactories = append(manifestFactories, manifests.Factory(ServiceMonitor))
  }
}
```

This is clearer imo.

Reply:
Thx, good catch!